src/conn.rs

```diff
@@ -408,18 +410,26 @@ impl Conn {
             Packet::UserauthRequest(p) => {
                 if let ClientServer::Server(serv) = &mut self.cliserv {
                     let sess_id = self.sess_id.as_ref().trap()?;
-                    disp.event = serv.auth.request(sess_id, s, p)?;
+                    match p.method {
+                        AuthMethod::None | AuthMethod::Unknown(_) => {
+                            trace!("UserauthRequest None or Unknown method requested. sending a fail");
+                            serv.auth.resume_request(false, s)?;
+                        }
+                        _ => {
```

Comment:
Specifics of auth handling should be in servauth.rs, not the dispatch loop here.

A "None" authmethod is allowed to return success - for example if you want to allow login without any password, and drop to some kind of restricted shell. The picow demo does that currently https://github.com/mkj/sunset/blob/main/embassy/demos/common/src/server.rs#L175

Is your code handling it the same way?

I will run a rustfmt pass over the repo - will make these diffs easier to read.